src/tools/illink/test/Mono.Linker.Tests.Cases/DataFlow/VirtualMethodHierarchyDataflowAnnotationValidation.cs

```diff
@@ -41,7 +41,7 @@ public static void Main()
             RequirePublicMethods(typeof(ImplementationClass));
             RequirePublicMethods(typeof(IBaseImplementedInterface));
             RequirePublicMethods(typeof(BaseImplementsInterfaceViaDerived));
-            RequirePublicMethods(typeof(DerivedWithInterfaceImplementedByBase));
+            RequirePublicMethodsAndConstructor(typeof(DerivedWithInterfaceImplementedByBase));
```

Comment:
@sbomer could you have a look at the changes to the trimming tests? Pretty much all changes are related to ILLink sometimes treating `typeof` as an allocation too. For example:

```csharp
[DAM(All)]
class Foo {}

Foo f = null;
_ = f.GetType();
_ = typeof(Foo);
```

Is enough for ILLink to apply DAM to Foo even though it would never be returned from the `object.GetType` call. We enshrined this behavior into lots of tests so I had to add lots of allocations.

Reply:
> Would you mind opening an issue to track this?

Filed as https://github.com/dotnet/runtime/issues/118427. Turns out there's a bit more to it, the issue has a repro.